app/mutations/Manager/deleteManagerFromRevision.ts

```diff
@@ -32,16 +53,116 @@ export const mutation = graphql`
   }
 `;
 
-export const useDeleteManagerFromRevisionMutation = (
+// Delete mutation for a form_change that was created in a previous revision & should be updated with operation = 'ARCHIVE'.
+export const archiveMutation = graphql`
+  mutation deleteManagerFromRevisionWithArchiveMutation(
+    $input: UpdateFormChangeInput!
+    $projectRevision: ID!
+  ) {
+    updateFormChange(input: $input) {
+      formChange {
+        id
+        newFormData
+      }
+      query {
+        projectRevision(id: $projectRevision) {
+          managerFormChanges: projectManagerFormChangesByLabel {
+            edges {
+              node {
+                projectManagerLabel {
+                  id
+                  rowId
+                  label
+                }
+                formChange {
+                  id
+                  newFormData
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+`;
+
+const useDeleteMutation = (
+  commitMutationFn?: (
+    environment: Environment,
+    config: MutationConfig<
+      | deleteManagerFromRevisionMutation
+      | deleteManagerFromRevisionWithArchiveMutation
+    >
+  ) => Disposable
+) => {
+  return useMutation<
+    | deleteManagerFromRevisionMutation
+    | deleteManagerFromRevisionWithArchiveMutation
+  >(deleteMutation, commitMutationFn);
+};
+
+const useArchiveMutation = (
   commitMutationFn?: (
     environment: Environment,
-    config: MutationConfig<deleteManagerFromRevisionMutation>
+    config: MutationConfig<
+      | deleteManagerFromRevisionMutation
+      | deleteManagerFromRevisionWithArchiveMutation
+    >
   ) => Disposable
 ) => {
-  return useMutation<deleteManagerFromRevisionMutation>(
-    mutation,
-    commitMutationFn
-  );
+  return useMutation<
+    | deleteManagerFromRevisionMutation
+    | deleteManagerFromRevisionWithArchiveMutation
+  >(archiveMutation, commitMutationFn);
+};
+
+export const useDeleteManagerFromRevisionMutation = (): [
+  (options: DeleteManagerOptions) => void,
+  boolean
+] => {
+  const [deleteManager, isDeleting] = useDeleteMutation();
+  const [archiveManager, isArchiving] = useArchiveMutation();
+  const deleteOrArchiveManager = (options: DeleteManagerOptions) => {
+    const { context, onCompleted, onError } = options;
+    if (context.operation === "CREATE") {
```

Comment:
This logic was moved away from the component, but is still a duplicate of `app/hooks/useDiscardFormChange.ts`. Can't this be factorized?

Reply:
Possibly? I was trying to ensure the nice generic hook stays generic & handle this specific case where the mutation requires a query in the return separately. I think `deleteManagerFromRevision` could maybe call `useDiscardFormChange` if `useDiscardFormChange` was refactored to take custom delete/archive mutations and a projectRevisionId as parameters. But then we'd need to add logic specific to the deletion of managers to the generic hook's code, which I was trying to avoid. This does cause duplication though.

Reply:
Let's leave it as is and see if we want to refactor later